pan3d/viewer/engine.py

```diff
@@ -139,15 +150,30 @@ def __init__(self, server, mesher):
 
     @vuwrap
     def reset(self, **kwargs):
-        self.mesher.algorithm.Modified()
-        self.plotter.clear()
-        self.actor = self.plotter.add_mesh(
-            self.mesher.algorithm,
-            show_edges=self._state.view_edge_visiblity,
-            clim=self.mesher.data_range,
-            **kwargs
-        )
-        self.plotter.view_isometric()
+        if not self._state.array_active:
+            return
+        try:
+            try:
+                # check for valid data
+                self.mesher.algorithm.RequestData(None, None, None)
```

Comment:
Never call `RequestData()` directly on any VTK algorithm. Simply call `Update()`

Reply:
Thank you! I had been looking for a function like that! However, it doesn't seem to raise any exceptions for us to show to the user when the update fails. For example, if the user selects one axis, the `RequestData` function was able to give us an error saying more axes are needed for a render. Is there some other function we can call to get such errors to the user's eyes?

Reply:
> However, it doesn't seem to raise any exceptions for us to show to the user when the update fails

This is because `RequestData()` is called by the VTK pipeline directly and error handling is managed on the VTK c++ side. This is why I have the `try/except` block that does something with the error message directly.

> For example, if the user selects one axis, the RequestData function was able to give us an error saying more axes are needed for a render.

Perhaps we should do this check outside of the algorithm to make sure the algorithm's inputs are valid at first

> Is there some other function ...

I think these are the errors you are dealing with:

https://github.com/pyvista/pyvista-xarray/blob/2c76baa986f30ea946cdf5cf51628f54643b35ef/pvxarray/rectilinear.py#L49-L57

Perhaps there's a way to implement a similar check here

Reply:
It works if I add these lines to the `validate_mesh` function: [7b56cf0](https://github.com/Kitware/pan3d/pull/23/commits/7b56cf004cdcc38542aa03d9f660ae09011ba37e). Is this good design for fixing the problem?